src/lib.rs

```diff
@@ -266,18 +266,33 @@ pub mod util {
     }
 
     pub fn load_private_key(filename: &str) -> Result<rustls::PrivateKey> {
-        let keyfile = fs::File::open(filename).map_err(|e| Error::Io(e))?;
+        use std::io::Seek;
+        use std::io::BufRead;
+
+        let keyfile = fs::File::open(filename).map_err(Error::Io)?;
         let mut reader = BufReader::new(keyfile);
-        let mut keys = pemfile::rsa_private_keys(&mut reader)
-            .map_err(|_| Error::BadKey)?;
 
-        // Ensure there's only one key in the file.
-        if keys.len() != 1 {
-            return Err(Error::BadKeyCount);
-        }
+        let private_keys_fn = {
+            let mut line = String::new();
+            reader.read_line(&mut line).map_err(Error::Io)?;
+            let private_keys_fn = match line.trim_right() {
+                "-----BEGIN RSA PRIVATE KEY-----" => pemfile::rsa_private_keys,
+                "-----BEGIN PRIVATE KEY-----" => pemfile::pkcs8_private_keys,
+                _ => return Err(Error::BadKey)
+            };
+            reader.seek(io::SeekFrom::Start(0)).map_err(Error::Io)?;
```

Comment:
I actually went a couple steps further and that part, making another nested block and making the match statement the final expression for `private_keys_fn`. Sorry about that, if it's too much I can change it back to what it was.